### PR TITLE
Scratch Changes & Code Clean-Up

### PR DESCRIPTION
**Pull request by Jl5her**
# Description 

### Backend Changes
- Scratch API proxy for projects and users (CORS workaround)
- Validate scratch project template is accessible when creating gold problems
- Problem IDs can be the same if they are different divisions
- Validate scratch project is accessible when creating gold submission.
- Allow team's to change their scratch_username (Protect against username, displayname, division, role, and school. Leaves password open to be changed potentially)

#### Components
- DivisionLabel  `<DivisionLabel division='blue | gold | public'/>`
- ScratchViewer `<ScratchViewer projet_id/>`
- StatusMessage `<StatusMessage message={type: 'error' | 'warning' | 'success' | undefined, message: string} />`
- ProblemEditor `<ProblemEditor problem? setProblem? handleSubmit={Problem => void} />`
- PageLoading `<PageLoading/>` - Shows standard loader for pages

#### Other Changes
- Admin problems page uses memoization for filtering problems by division
- Admin submissions page show's all submissions by default
- Updated Gold Problem page to new standards (Loader, Mounted)
- Includes link to Scratch project
- Updated Gold Problems view to new standards (Loader, Mounted)
- Update Gold submit page.
- Shows ScratchViewer for project_url
- Allows passing default problem 
- _cpu_time_limit_, _memoy_limit_, _tests_, and _project_id_ are optional

## Type of change

- [x] Tested individually.

# Checklist:

- [x] My code follows the style guidelines of this project.
- [x] I have performed a self-review of my own code.
- [x] I have commented my code, particularly in hard-to-understand areas.
- [x] My changes generate no new warnings.
- [x] My changes do not break any features.

### CODE REVIEW

File: frontend/src/pages/About.tsx [hunk 1]
```diff
@@ -24,38 +22,17 @@ const About = (): JSX.Element => (
...
+        Teams of two or three students will have three hours and one computer to implement a themed Scratch project. Submissions will be judged on creativity, use of key Scratch constructs, and theme correspondence. This division will be suitable for introductory students familiar with Scratch.
       </p>
 
       <h2>AP Computer Science Principles (Eagle)</h2>
       <p>
-        Teams of two to four students will be working together to solve a problem that is present in society and is awaiting a technological
-        solution. The students then have three hours to develop a solution using their knowledge of computer science principles and technologies. Students are
-        not required to write code or create a working prototype, but rather have a flushed out, technical solution. At the end of the three hours, each team will
-        present (5 – 10 minutes) their solution to a small board of faculty members. The faculty will ask a few questions and ultimately vote on a winner. We will
-        have a Google Meet call where students will present and gain feedback the feedback mentioned earlier.
```
Comment (brad-cooley) [inline frontend/src/pages/About.tsx:58]: We want to keep the previous changes here.

File: frontend/src/pages/admin/submissions/Submissions.tsx [hunk 2]
```diff
@@ -20,9 +21,7 @@ const Submissions = (): JSX.Element => {
   const [isLoading, setLoading] = useState<boolean>(true)
   const [submissions, setSubmissions] = useState<SubmissionItem[]>([])
   const [isMounted, setMounted] = useState<boolean>(true)
-  const [showReleased, setShowReleased] = useState(false)
-
-  const helmet = <Helmet><title>Abacus | Admin Submissions</title></Helmet>
+  const [showReleased, setShowReleased] = useState(true)
```
Comment (brad-cooley) [inline frontend/src/pages/admin/submissions/Submissions.tsx:24]: Do we want this to always be true? When competition time comes, that page is going to be bloated if we show released. I think we should just show unreleased.

File: frontend/src/pages/gold/Problems.tsx [hunk 3]
```diff
@@ -2,73 +2,80 @@ import { Problem } from 'abacus'
...
+        <Table.Header>
+          <Table.Row>
+            <Table.HeaderCell />
+            <Table.HeaderCell>Problem Name</Table.HeaderCell>
+          </Table.Row>
+        </Table.Header>
+        <Table.Body>
+          {!(problems?.length) ?
+            <Table.Row>
+              <Table.Cell colspan={2} style={{ textAlign: 'center' }}>We can&apos;t find any problems. If you believe this is an error please contact us.</Table.Cell>
```
Comment (brad-cooley) [inline frontend/src/pages/gold/Problems.tsx:65]: want to make `colSpan={'100%'}`?

File: frontend/src/pages/gold/Standings.tsx [hunk 4]
```diff
@@ -6,18 +6,16 @@ import { Helmet } from 'react-helmet';
...
-      </Helmet>
+      <Helmet> <title>Abacus | Gold Standings</title> </Helmet>
       <Countdown />
       <Block center size='xs-12'>
         <h1>Competition not yet started!</h1>
         <p>Standings will become available when the competition begins, and submissions start rolling in.</p>
       </Block>
     </>
-  }
+
```
Comment (brad-cooley) [inline frontend/src/pages/gold/Standings.tsx:18]: So we have no table for standings if the competition is started?

**Review by brad-cooley — CHANGES_REQUESTED** [review]

File: frontend/src/pages/admin/submissions/Submissions.tsx (hunk 2 above)
Comment (Jl5her) [reply frontend/src/pages/admin/submissions/Submissions.tsx:24]: Keep it only showing unreleased until pagination is implemented?

File: frontend/src/pages/gold/Standings.tsx (hunk 4 above)
Comment (Jl5her) [reply frontend/src/pages/gold/Standings.tsx:18]: Not yet

File: frontend/src/pages/gold/Problems.tsx (hunk 3 above)
Comment (Jl5her) [reply frontend/src/pages/gold/Problems.tsx]: https://github.com/acm-mu/abacus/pull/109/commits/92a8171e822ec7218a69d660013741cf410dde98

File: frontend/src/pages/About.tsx (hunk 1 above)
Comment (Jl5her) [reply frontend/src/pages/About.tsx]: Is this right now, https://github.com/acm-mu/abacus/commit/934d1916f6ca13ac43717f5e74854680e8391f1a?

File: frontend/src/pages/admin/submissions/Submissions.tsx (hunk 2 above)
Comment (Jl5her) [reply frontend/src/pages/admin/submissions/Submissions.tsx]: https://github.com/acm-mu/abacus/pull/109/commits/12a9b5ab6215591dacf03b19d3beebd671e80c85 changed.

**Review by brad-cooley — APPROVED** [review]

**Pull request merged**